first_order_model/shrink_util.py

```diff
@@ -951,82 +1014,83 @@ def compute_deletion(layer_graph,
     for after_layer in layer_graph[layer].after:
         follow(following_layers, layer_graph, layer_graph[after_layer])
 
-    # If you delete a part of the previous one, not just all of it then you cannot just say these n are to be deleted
-    # So for each layer you need to figure out what you are deleting from it, then for the next layer figure out what is deleted
-
     deletions = {}
     if isinstance(custom, list):
-        deletions[layer] = custom
+        deletions[layer] = copy.copy(custom)
     elif isinstance(custom, int):
         # The sorting magic happens here
         # Find the first convolution
         if sort:
-            if layer_graph[layer].type == 'bn':
-                breakpoint()
+            assert layer_graph[layer].type != 'bn', "We cant sort batchnorm layers"
+
+            # Get the layer which accepts this ones output as input.
             depthwise_skipped_following_layers = []
             for after_layer in layer_graph[layer].after:
                 get_following_layers_skip_depthwise(depthwise_skipped_following_layers, layer_graph, layer_graph[after_layer])
 
             first_conv = get_first_conv(layer_graph, depthwise_skipped_following_layers)
 
-            if first_conv is None:
-                breakpoint()
+            assert first_conv is not None, "First conv is none"
 
-            slices = get_relevent_slice(layer_graph, first_conv, layer)
-            if len(slices) == 0:
-                breakpoint()
+            # Find the relevant slices of input which matters if there is a concat in the way
+            slices = get_relevant_slice(layer_graph, first_conv, layer)
 
-            importances = torch.zeros(layer_graph[layer].o).cuda()
+            assert len(slices) != 0, "Slices length is 0"
+
+            # Generate importances going slice by slice of the next layer..
+            importances = torch.zeros(layer_graph[layer].out_channels).cuda()
             for single_slice in slices:
                 importances += get_importances(layer_graph[first_conv].value.weight, single_slice)
```

Comment:
I'm not sure what's going on with this importance list vs. computing importance slice by slice and adding it? can you explain why you're using += on a tensor without indexing?

Reply:
I think my new comment explains it.